### PR TITLE
Search fixes

### PR DESCRIPTION
**Pull request by Waqar144**
- Rewrote `highlightSearch` to use the available API
- Focus on textEdit on return key press

#239

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -2817,34 +2827,21 @@ void MainWindow::highlightSearch() const
...
-        }
+    m_textEdit->moveCursor(QTextCursor::Start);
 
-        cursor.endEditBlock();
+    QList<QTextEdit::ExtraSelection> extraSelections;
+    while(m_textEdit->find(searchString)) {
+        QTextEdit::ExtraSelection extra = QTextEdit::ExtraSelection();
+        extra.format.setBackground(Qt::yellow);
+        extra.cursor = m_textEdit->textCursor();
+        extraSelections.append(extra);
```
Comment (bjorn) [inline src/mainwindow.cpp]: The format is always the same, so we can share it:

```c++
QTextCharFormat highlightFormat;
highlightFormat.setBackground(Qt::yellow);

QList<QTextEdit::ExtraSelection> extraSelections;
while(m_textEdit->find(searchString))
    extraSelections.append({ highlightFormat, m_textEdit->textCursor() });

**Comment by nuttyartist** [comment]
Two things:

1. Ctrl-enter doesn't seem to work on my computer (Mac) while searching, it doesn't focus and show the cursor on the selected note. Maybe it's a problem specific to Mac or my computer.
2. It doesn't relate to this PR, but maybe you can solve it here. It seems like the selected note color doesn't change when a user enter into the note. It should change to m_ActiveColor(218, 233, 239) from notewidgetdelegate.cpp. Can you solve it?

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -2743,6 +2743,20 @@ bool MainWindow::eventFilter(QObject *object, QEvent *event)
...
+        auto *keyEvent = static_cast<QKeyEvent *>(event);
+        if (keyEvent->key() == Qt::Key_Return && m_searchEdit->text().isEmpty()) {
+            setFocusOnText();
+            return true;
+        } else if (keyEvent->key() == Qt::Key_Return &&
+                   keyEvent->modifiers().testFlag(Qt::ControlModifier)) {
+            setFocusOnText();
+            return true;
+        }
+        return QMainWindow::eventFilter(object, event);
```
Comment (bjorn) [inline src/mainwindow.cpp]: Why is this line here? I guess it could just be removed because there is already `return QObject::eventFilter(object, event);` below (which could also use `QMainWindow`, but `QMainWindow` and `QWidget` don't override this virtual anyway).

File: src/mainwindow.cpp (hunk 1 above)
Comment (Waqar144) [reply src/mainwindow.cpp]: Done

**Comment by Waqar144** [comment]
> Two things:
> 
>     1. Ctrl-enter doesn't seem to work on my computer (Mac) while searching, it doesn't focus and show the cursor on the selected note. Maybe it's a problem specific to Mac or my computer.
> 
>     2. It doesn't relate to this PR, but maybe you can solve it here. It seems like the selected note color doesn't change when a user enter into the note. It should change to m_ActiveColor(218, 233, 239) from notewidgetdelegate.cpp. Can you solve it?

Both done.

File: src/mainwindow.cpp (hunk 2 above)
Comment (Waqar144) [reply src/mainwindow.cpp]: Ah, yes. I noticed it later. I will just convert into `break;` 
thanks

**Comment by nuttyartist** [comment]
Some problems:

1. Ctrl-Enter while searching doesn't focus on textEdit and put the cursor there for me, it
2. When searching and focusing on textEdit using the mouse (currently selected note content), it doesn't change the color to m_ActiveColor for some reason. But it does when using Ctrl-Enter (although, as stated in 1, it doesn't put the cursor there and focus).

**Comment by Waqar144** [comment]
Ctrl + Enter is working for me on Linux 🤔

**Comment by Waqar144** [comment]
The mousePressEvent doesn't get caught in eventFilter for some reason. So yeah, that part doesn't work, with / without searching (clicking into textEdit).

**Comment by nuttyartist** [comment]
Hmm... Do you have an idea why?

And about Ctrl+Enter, it works also for me but doesn't put the cursor on textEdit while searching, only when not searching. But it still might be a Mac problem.

**Comment by nuttyartist** [comment]
Oh ok, it put the cursor next to the found text, just noticed. Works well! Merging...

Thanks @Waqar144!

**Comment by nuttyartist** [comment]
This PR appears to create a serious bug where pressing Enter doesn't work in a notes' content - therefore I can't create new lines. I created a new issue for this #244

**Comment by Waqar144** [comment]
> Hmm... Do you have an idea why?

Maybe because there is already an override for `mousePressEvent` in mainwindow.cpp. If we subclass QTextEdit and handle the mouse press event there, it might work. And that is something that is eventually going to have to happen, for Rich Text/ text editing commands etc.
